File: JAVA/level2/올바른_괄호.java
class Solution {
    boolean solution(String s) {
        char[] chars = s.toCharArray();
        int num = 0;

        for(char c : chars){
            if(c == '(')
                num++;
            if(c == ')')
                num--;

            if(num < 0)
                return false;
        }

        if(num != 0)
            return  false;
        return true;
    }
}

File: JavaScript/level2/올바른_괄호.js
function solution(s){
    let count = 0;
    
    for(let i of s){
        if(count < 0)
            return false;
        if(i === "(")
            count++;
        if(i === ")")
            count--;
    }
    
    if(count === 0)
        return true;
    
    return false;
}
